lib/drivers/node-mongodb-native/connection.js

```diff
@@ -88,6 +91,12 @@ NativeConnection.prototype.useDb = function(name) {
   this.otherDbs.push(newConn);
   newConn.otherDbs.push(this);
 
+  // push onto the relatedDbs stack, this is used when state changes
+  if (options.useCache) {
```

Comment:
One more detail that we missed, need to check if `options` is undefined here, that's why [the tests are failing](https://travis-ci.org/Automattic/mongoose/jobs/344515056)

Reply:
@rocketspacer you also need to apply your changes from 5560bd0 to this line :+1: